### ISSUE TITLE
Race condition leading to `Argument should belong to the same JavaScript world as target object`

### ISSUE THREAD

**Issue by aaronjensen** [rubycdp/cuprite#233]
We are occasionally seeing this error come from Ferrum:

```
Argument should belong to the same JavaScript world as target object
```

From what I can tell, it may have to do with how cuprite handles finding within elements. If the page navigates in the middle of a find (in between the time when the node ID is resolved in Ruby and when that node is sent to the browser) then the crash occurs.

**Comment by emilkarl** [comment]
Did you solve this?

**Comment by aaronjensen** [comment]
No, we have tended to avoid `within` for this reason. I haven't looked into it recently as a result.

**Comment by markedmondson** [comment]
Just started seeing this occurring frequently, but with no obvious changes. We do use `within` quite a lot as above but it had been fine for months/years...

**Comment by route** [comment]
@markedmondson something has changed on the js side maybe?

**Comment by route** [comment]
Intermittent issues like that often manifests themselves on CI due to its slowness. Lower memory ceiling, slower CPU and it fails more often in co parison to flying macbooks
